File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    var answer = '';
    
    const map = new Map();
    map.set('RT', {'R': 0, 'T': 0}); map.set('CF', {'C': 0, 'F': 0}); map.set('JM', {'J': 0, 'M': 0}); map.set('AN', {'A': 0, 'N': 0});
    
    for(let i=0; i<survey.length; i++){
        const type = survey[i].split('').sort().join('');
        const choice = choices[i];
        
        const obj = map.get(type);
        
        let first = type[0];
        let second = type[1];
        if(type[0] !== survey[i][0]) [first, second] = [second, first];
        
        if(choice<4)    obj[first] += 4-choice;
        else if(choice>4)   obj[second] += choice-4;
    }
    
    for(const [key, value] of map){
        const first = key[0];
        const second = key[1];
        
        if(value[first]>value[second]) answer+=first;
        else if(value[first]<value[second]) answer+=second;
        else    answer+=first;
    }
    
    console.log(map)
    
    return answer;
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
# survey를 for문 순환하면서 점수 업데이트
# types_A, types_B로 나눠서 타입에 맞는 점수 계산

def solution(survey, choices):
    answer = ''
    
    types = {type: 0 for type in ['R', 'T', 'C', 'F', 'J', 'M', 'A', 'N']}
    
    for type, choice in zip(survey, choices):
        A, B = list(type)
        
        if choice < 4:
            types[A] += 4 - choice
        elif choice > 4:
            types[B] += choice - 4
        
    
    answer += 'R' if types['R'] >= types['T'] else 'T'
    answer += 'C' if types['C'] >= types['F'] else 'F'
    answer += 'J' if types['J'] >= types['M'] else 'M'
    answer += 'A' if types['A'] >= types['N'] else 'N'
    
    return answer